src/plfa/part2/Untyped.lagda.md

```diff
@@ -721,6 +721,8 @@ This works because:
     ≡
       f · (μ f)
 
+This fixpoint definition is well-known as [the Y combinator](https://en.wikipedia.org/wiki/Fixed-point_combinator#Fixed-point_combinators_in_lambda_calculus), discovered by Haskell Curry.
```

Comment:
We probably shouldn't be referring to Wikipedia or saying Curry invented it without a source.

Reply:
OK. Shall I edit this?

Reply:
OK tried tracing its source but I couldn't get access to the content of the second citation in the Wikipedia page, "Henk Barendregt (1985). The Lambda Calculus – Its Syntax and Semantics. Studies in Logic and the Foundations of Mathematics. 103. Amsterdam: North-Holland. ISBN 0444867481.".

Reply:
When I said it should include a citation, I meant a citation to a book rather than a link to Wikipedia. (As Wen notes, links to Wikipedia are not proper academic citations.) It should be to the first person to name the combinator Y, which is not Barendregt.

I tried to track down the original citation just now, but failed. You can cite this paper, which is reasonably authorative, but doesn't actually say who first defined Y: http://www.users.waitrose.com/~hindley/SomePapers_PDFs/2006CarHin,HistlamRp.pdf. The footnote on the first page of the link gives the proper citation.